servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/ChannelSet.java

```diff
@@ -87,14 +97,17 @@ public ChannelSet(Executor offloadingExecutor) {
      */
     public boolean addIfAbsent(final Channel channel) {
         final boolean added = channelMap.putIfAbsent(channel.id(), channel) == null;
+        if (added) {
+            final ConnectionObserver observer = this.connectionObserverFactory != null ?
+                    this.connectionObserverFactory.get() : null;
+            if (observer != null) {
+                assignConnectionObserver(channel, observer);
```

Comment:
Can we do this assignment in `TcpServerBinder`? `ChannelSet` is just a means to have a collection of channels for a server; adding this functionality here seems incorrect. Additionally it increases the scope of `TransportObserverUtils` outside the tcp-netty-internal module

Reply:
Good call, let me move it from `ChannelSet` to a better place.